migrationtools/migrations.py

```diff
@@ -798,6 +798,74 @@ def migrate_helpdesk(self):
         }
         self.helpdesk.update_taskstate('contact-pull', task_data)
 
+    def migrate_helpdesk_stubbed(self):
+        ident_cols = self.seed_identity.identity.c
+        cont_cols = self.helpdesk.contact.c
+
+        for contact in self.helpdesk.get_stubbed_contacts():
+            # Keep a record of all transactions started so they can all be rolled back on any error.
+            transactions = {}
+
+            uid = contact[cont_cols.id]
+            self.echo("Starting fix of {id} ...".format(id=uid), nl=False)
+
+            uuid = contact[cont_cols.uuid]
+            identity = self.seed_identity.lookup_identity(uuid)
+            if identity is None:
+                self.echo(" Failed")
+                self.echo("Could not load identity for uuid : {0}".format(uuid), err=True)
+                continue
+
+            lang = identity[ident_cols.details].get('lang_code', 'eng_ZA')
+            # casepro is expecting 639-2 code
+            lang, _, _ = lang.partition('_')
+            addresses = identity[ident_cols.details]['addresses']
+            urns = []
+            for scheme, address in addresses.items():
+                scheme_addresses = []
+                for urn, details in address.items():
+                    if 'optedout' in details and details['optedout'] is True:
+                        # Skip opted out URNs
+                        continue
+                    if 'default' in details and details['default'] is True:
+                        # If a default is set for the scheme then only store the default
+                        scheme_addresses = [urn]
+                        break
+                    scheme_addresses.append(urn)
+                for value in scheme_addresses:
+                    if scheme == "msisdn":
+                        scheme = "tel"
+                    urns.append("%s:%s" % (scheme, value))
+
+            contact_details = {
+                'uuid': uuid,
+                'language': lang,
+                'org_id': 1,  # hard default to 1
+                'name': None,
+                'fields': {},
+                'urns': urns,
+                'is_blocked': False,
+                'is_stub': False,
+                'is_active': True,
+                'is_stopped': False,
+            }
```

Comment:
We could probably extract this conversion out into a function, so that it doesn't have to be repeated.